hfaerber/Palette_Picker
# Home component

### What's this PR do? 

- Adds and styles the home component. 
- Adds an api call to fetch projects. This api call is dynamic so that it can fetch all projects or a single project.
- Adds an api call for posting a project. 

----------

### How should this be manually tested? 

- `git fetch origin`
- `git checkout home-component`
- `npm start`
- Navigate to localhost:3000/home in your browser and make sure the page loads correctly. Try adding a project and make sure it is rendered to the page. 

----------

### Relevant tickets:

- #20 

----------

### Screenshots: 

<details>

<summary>Home in browser</summary>

 ![ ](https://user-images.githubusercontent.com/47998896/74089578-d2f76a80-4a5f-11ea-950f-1630e109da9d.png)

</details>

----------

### Notes: 

- This PR does not include a test file. I will add more cards to the project board for testing components. 
- Try adding a project without entering a name to see the error message. I'm not sure I like the styling so please let me know if you have any suggestions.
- See my note on the extra fetching in the notes/questions columns on the project board. Everything works so I'm not sure if it's worth the effort to change the endpoint around. 
- We need to add some conditional rendering so that you are redirected to `/`. Right now, refreshing the page removes the user from App's state and no username appears on the page.
